File: apps/backend/identity-service/identity-domain/identity-application-service/src/main/java/org/sabha/identity/applicationservice/WebSession.java
package org.sabha.identity.applicationservice;

import java.util.Set;

import org.sabha.identity.domain.Section;

/**
 * What the web shell needs to render after login (Slice 9): the User's display
 * name, whether they are a Madhyastha Karyalaya member, and the set of
 * {@link Section}s their authority unlocks.
 */
public record WebSession(String username, boolean madhyasthaKaryalaya, Set<Section> sections) {
}

File: apps/web/projects/identity-domain/src/lib/web-session.ts
import { Section } from './section';

/**
 * What the web shell needs to render after login (Slice 9): the User's display
 * name, whether they are a Madhyastha Karyalaya or Regional Team member, and the
 * {@link Section}s their authority unlocks. Frontend mirror of the backend
 * `WebSession` record; the JSON shape matches
 * `BffSessionController.WebSessionResponse`.
 */
export interface WebSession {
  username: string;
  madhyasthaKaryalaya: boolean;
  regionalTeam: boolean;
  sections: Section[];
}
